govcd/disk_test.go

```diff
@@ -46,16 +46,15 @@ func (vcd *TestVCD) Test_CreateDisk(check *C) {
 	check.Assert(task.Task.Owner.Type, Equals, types.MimeDisk)
 	diskHREF := task.Task.Owner.HREF
 
-	// Defer prepend the disk info to cleanup list until the function returns
-	defer PrependToCleanupList(fmt.Sprintf("%s|%s", diskCreateParamsDisk.Name, diskHREF), "disk", "", check.TestName())
+	AddToCleanupList(diskCreateParamsDisk.Name, "disk", diskHREF, check.TestName())
```

Comment:
Two problems:
1. Why changing `defer Prepend` to `Add`? There is a reason for this usage, which becomes clear when running the whole test suite.
2. I don't like this change from the previous usage. We are using the `parent` field to store the HREF.
If we need a more complex structure, let's propose it, but using a field for what is not will make the code harder to understand.

Reply:
1. To be consistency as we use `Add` everywhere and `prepend` looked suspicious regarding behaviour against `Add`
2. Yes you are right in that sense. I can change it to `AddToCleanupList(diskHREF, "disk", "", check.TestName())`

Reply:
updated